packages/macro/src/index.macro.ts

```diff
@@ -135,23 +104,29 @@ const macro: MacroHandler = ({ references, state }) => {
       if (!t.isObjectExpression(optionsNode)) {
         throw new Error("The third argument must be an options object.");
       }
-      const options = parseToOptions(optionsNode);
+      const macroOptions = extractMacroOptions(optionsNode);
 
-      const webpackChunkName = (options.webpackChunkName ? options.webpackChunkName : 'default') + '-ssr-computations';
+      const webpackChunkName = (macroOptions.webpackChunkName ? macroOptions.webpackChunkName : 'default') + '-ssr-computations';
 
       const absolutePath = path.resolve(
         path.dirname(currentFilename),
         filenameNode.value,
       );
 
+      if (macroOptions.serverSideOnly && side !== 'server') {
+        // replace the call to useSSRComputation with null
+        nodePath.parentPath?.replaceWith(t.nullLiteral());
+        return;
```

Comment:
This option will result in hydration conflicts.
For successful hydrations, ALL the variables in rendering functions must have exactly the same value client-side and server-side. [React relies on this](https://react.dev/reference/react-dom/client/hydrateRoot#caveats).

Reply:
@Romex91 I updated the server-side code to be `(() => null)(useSSRComputationFunction(....))`. So, the server-side will also return `null`. It should be used only when we want to cache a specific value to be used later. I mean, we don't want to use this value now, but we will need it in another place.

Reply:
The name `serverSideOnly` doesn't exactly correspond to what the feature does since it affects the client too.

Isn't `noReturnValue` a better name?

Reply:
Also, could you please describe a usage example? Why do you want to introduce this option?

Reply:
@Romex91 It can be used when we need to cache a specific value without needing its value right now. It's supposed to be used when we need to cache a value that will be needed in a Component that will not be server-side rendered.
For example, let's assume we have the following components:
- `Main Page`: contains the page's main layout and is server-side rendered.
- `UserProfile`: displays the user info. The `MainPage` uses the `UserProfile` component, but it lazily loads it. So, the `UserProfile` will not be server-side rendered.
Assume that the `UserProfile` component needs to query the user info. However, the `MainPage` itself doesn't need the user info.

Using the `serverSideOnly` option, we can run the computation to cache the user info query inside the `MainPage`. The purpose of running the computation inside the `MainPage` is to cache the info only without using them. The caching is done on the server side only, so we don't need to include the call in the consumer bundle of the `MainApp`.

**By using this option the `useSSRComputation.runtime` package will be only bundled in side the client bundle of the `UserProfile` component instead of being bundled inside the `MainPage` component.

Reply:
@Romex91 Done, remove the `serverSideOnly` option completely